Sources/Foundation/JSONSerialization.swift

```diff
@@ -737,3 +701,48 @@ private extension JSONValue {
         }
     }
 }
+
+extension NSNumber {
+    static func fromJSONNumber(_ string: String) -> NSNumber? {
```

Comment:
This should probably be a constructor (`init?(jsonNumberString string: String)`).

Reply:
In some cases this might return a `NSDecimalNumber`. I don't think that we can make it work with a `convenience` initializer (which is the only option we have in an extension).

Reply:
Yeah seems to be the case even though `NSDecimalNumber` is a subclass of `NSNumber`